pkg/sessions/code.go

```diff
@@ -0,0 +1,65 @@
+package sessions
+
+import (
+	"time"
+
+	"github.com/cozy/cozy-stack/pkg/crypto"
+)
+
+// CodeLen is the number of random bytes used for the session codes
+const CodeLen = 16
+
+// A Code is used to transfer a session from the domain with the stack to a
+// subdomain of an application, when the flat subdomains structure is used.
+// The code is valid:
+//   - only once
+//   - for a short time span (1 minute)
+//   - just for one application.
+type Code struct {
+	Value     string
+	SessionID string
+	AppHost   string
+	ExpiresAt int64
+}
+
+// The list of pending codes
+var codes []*Code
```

Comment:
Nope, it's really better from a security point of view to make sure that a code can't be reused, and this is only possible with a state on the server.

At scale, we will use redis for them (and it will be a constant time check).